crates/nargo/src/cli/verify_cmd.rs

```diff
@@ -44,12 +44,15 @@ pub(crate) fn run(args: VerifyCommand, config: NargoConfig) -> Result<(), CliErr
 
     let result = verify_with_path(
         config.program_dir,
-        proof_path,
+        proof_path.clone(),
         circuit_build_path,
         false,
         args.allow_warnings,
     )?;
-    println!("Proof verified : {result}");
+
+    if !result {
+        return Err(CliError::ProvingFailed(proof_path));
```

Comment:
It's not the proving that failed here but verification so this name seems a little off.

```suggestion
        return Err(CliError::InvalidProof(proof_path));
```